File: python/arranging-coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        left, right = 1, n
        while left <= right:
            mid = (left + right) // 2
            complete_row_coins = mid * (mid + 1) / 2
            if n == complete_row_coins:
                return mid
            if n < complete_row_coins:
                right = mid - 1
            else:
                left = mid + 1
        return right

File: rust/arranging-coins.rs
impl Solution {
    pub fn arrange_coins(n: i32) -> i32 {
        let mut left = 1;
        let mut right = n;
        while left <= right {
            let mid = (right + left) / 2;
            let complete_rows_coins = mid * (mid + 1) / 2;
            if n == complete_rows_coins {
                return mid;
            } else if n < complete_rows_coins {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return right;
    }
}
